datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -1198,6 +1199,11 @@ def run(self):
         # type: () -> str
         try:
             self.diagnosis.clear()
+            if is_affirmative(datadog_agent.get_config("integration_profiling")):
+                Profiling().start()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Do we need to stop the profiling at the end of each run or should we let it run in between runs (with a ~15sec sleep)?

Reply:
While the `integration_profiling` configuration is enabled the profiler would run between runs.  Only if between a run that the agent has changed the `integration_profiling` value that we would be disable. 

Did this answer your question?